pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -3589,9 +3589,16 @@ func (c *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, apiS
 		}
 	}
 
+	annotationProtocol := strings.ToLower(annotations[ServiceAnnotationLoadBalancerBEProtocol])
+	var hcProtocol string
 	if path, healthCheckNodePort := service.GetServiceHealthCheckPathPort(apiService); path != "" {
 		klog.V(4).Infof("service %v (%v) needs health checks on :%d%s)", apiService.Name, loadBalancerName, healthCheckNodePort, path)
-		err = c.ensureLoadBalancerHealthCheck(loadBalancer, "HTTP", healthCheckNodePort, path, annotations)
+		if annotationProtocol == "https" || annotationProtocol == "ssl" {
```

Comment:
This won't work, since the healthCheck for externalTrafficPolicy==Local(which triggers this branch) is in HTTP, no matter what ServiceAnnotationLoadBalancerBEProtocol is

Reply:
Thanks for letting me know. So I can just check `if (annotationProtocol == "https" || annotationProtocol == "ssl") && apiService.Spec.ExternalTrafficPolicy != api.ServiceExternalTrafficPolicyTypeLocal` and then I'll be good?

Reply:
Hi, you can just leave this branch untouched(i.e. still pass "HTTP" as protocol"), since this branch is only triggered when externalTrafficPolicy==Local, see https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/service/util.go#L88

Reply:
FYI, The healthCheck listener is allocated by k8s on nodes for externalTrafficPolicy==Local as HTTP, which returns 200 on nodes with pods of that service and 503 vice versa, see https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/proxy/healthcheck/healthcheck.go#L177